frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -1,17 +1,40 @@
+import * as _ from 'lodash-es';
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import { Heading } from './utils';
+import { Heading, CopyToClipboard } from './utils';
 
-export const ConfigMapData = ({data}) => {
+export const MaskedData = () => {
+  return <React.Fragment>
+    <span className="sr-only">value hidden</span>
+    <span aria-hidden="true">&bull;&bull;&bull;&bull;&bull;</span>
+  </React.Fragment>;
+};
+
+export const KeyValueData = ({data, showSecret}) => {
   const dl = [];
   Object.keys(data || {}).sort().forEach(k => {
+    let value = '';
+    let visibleValue = '';
+    if (_.isNil(showSecret)) {
+      value = visibleValue = data[k];
+    } else {
+      value = window.atob(data[k]);
+      visibleValue = showSecret ? value : <MaskedData /> ;
+    }
     dl.push(<dt key={`${k}-k`}>{k}</dt>);
-    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{data[k]}</pre></dd>);
+    dl.push(<dd key={`${k}-v`}><CopyToClipboard value={value} visibleValue={visibleValue}/></dd>);
   });
-
   return <dl>{dl}</dl>;
 };
 
+export const ConfigMapData = ({data}) => {
+  return <KeyValueData data={data}/>;
+};
+
+export const Icon = ({showSecret}) => {
+  return showSecret ? <span className="show-values"><i className="fa fa-eye-slash"></i></span> : <span className="hide-values"><i className="fa fa-eye"></i></span>;
```

Comment:
Missing aria-hidden

Reply:
In the mocks, the icon wasn't blue.

Reply:
Checked and the icon + link are blue only when not focused

Reply:
> Checked and the icon + link are blue only when not focused

Sorry, I'm not sure I understand. The icon shouldn't be blue when not focused, at least per the original mocks.